Lombiq.Tests.UI/MonkeyTesting/MonkeyTester.cs

```diff
@@ -0,0 +1,245 @@
+            return pageTestInfo != null;
+        }
+
+        private PageMonkeyTestInfo GetCurrentPageTestInfo()
+        {
+            var url = _context.Driver.Url;
+            var cleanUrl = CleanUrl(url);
+
+            var pageTestInfo = _pageTestInfoList.FirstOrDefault(x => x.CleanUrl == cleanUrl)
+                ?? new PageMonkeyTestInfo(url, cleanUrl, _options.PageTestTime);
+
+            Log.Info($"Current page is \"{pageTestInfo.CleanUrl}\"");
+
+            return pageTestInfo;
+        }
+
+        private string CleanUrl(string url)
+        {
+            foreach (var cleaner in _options.UrlCleaners)
+                url = cleaner.Handle(url, _context);
+
+            return url;
+        }
+
+        private void TestCurrentPage(PageMonkeyTestInfo pageTestInfo)
+        {
+            int randomSeed = GetRandomSeed();
+
+            TestCurrentPageWithRandomSeed(pageTestInfo, randomSeed);
+        }
+
+        private void TestCurrentPageWithRandomSeed(PageMonkeyTestInfo pageTestInfo, int randomSeed)
+        {
+            try
+            {
+                Log.ExecuteSection(
+                    new LogSection(
+#pragma warning disable S103 // Lines should not be too long
+                       $"Monkey test \"{pageTestInfo.CleanUrl}\" within {pageTestInfo.TimeToTest.ToShortIntervalString()} with {randomSeed} random seed"),
+#pragma warning restore S103 // Lines should not be too long
+                    () =>
+                    {
+                        ExecutePreAssertions();
+
+                        var pageTestTimeLeft = TestCurrentPageAndMeasureTestTimeLeft(pageTestInfo.TimeToTest, randomSeed);
+                        pageTestInfo.TimeToTest = pageTestTimeLeft;
+                        if (!_pageTestInfoList.Contains(pageTestInfo))
+                            _pageTestInfoList.Add(pageTestInfo);
+
+                        ExecutePostAssertions();
+                    });
+            }
+            catch (Exception exception)
+            {
+                throw new AssertionException($"Failure on \"{pageTestInfo.CleanUrl}\" page", exception);
+            }
+        }
+
+        private void ExecutePreAssertions()
+        {
+            if (_options.RunAccessibilityCheckingAssertion) _context.AssertAccessibility();
+
+            if (_options.RunHtmlValidationAssertion) _context.AssertHtmlValidityAsync().GetAwaiter().GetResult();
+        }
+
+        private void ExecutePostAssertions()
+        {
+            if (_options.RunAppLogAssertion) _context.AssertAppLogAsync().GetAwaiter().GetResult();
+
+            if (_options.RunBrowserLogAssertion) _context.AssertBrowserLogAsync().GetAwaiter().GetResult();
+        }
+
+        [SuppressMessage("Security", "SCS0005:Weak random number generator.", Justification = "For current purpose it should not be secured.")]
+        [SuppressMessage("Security", "CA5394:Do not use insecure randomness", Justification = "For current purpose it should not be secured.")]
```

Comment:
`NonSecurityRandomizer` doesn't have a constructor with seed parameter, which I need. Should I add it to `NonSecurityRandomizer`? It will also need another PR for Lombiq/Helpful-Libraries  repo.

Reply:
Right. For the sake of simplicity I added it.

Reply:
Applied `NonSecurityRandomizer`.